v2/streaming-data-generator/src/test/java/com/google/cloud/teleport/v2/templates/StreamingDataGeneratorTest.java

```diff
@@ -45,103 +34,167 @@
+                            assertThat(message.getPayload(), is(notNullValue()));
+                            assertThat(message.getAttributeMap(), is(notNullValue()));
+
+                            return null;
+                        });
+
+        pipeline.run();
     }
 
-    return resourceId;
-  }
+    /**
+     * Tests the {@link MessageGeneratorFn} does not fail when given invalid schema.
+     */
+    @Test
+    public void testMessageGeneratorInvalidSchema() throws IOException {
+        // Arrange
+        //
+        String schema = "{\"name: \"Invalid\"";
+
+        File file = tempFolder.newFile();
+        writeToFile(file.getAbsolutePath(), schema);
+
+        // Act
+        //
+        PCollection<PubsubMessage> results =
+                pipeline
+                        .apply("CreateInput", Create.of(0L))
+                        .apply("GenerateMessage", ParDo.of(new MessageGeneratorFn(file.getAbsolutePath())));
+
+        // Assert
+        //
+        PAssert.that(results).satisfies(input -> {
+            PubsubMessage message = input.iterator().next();
+            assertThat(message, is(notNullValue()));
+            assertThat(new String(message.getPayload()), is(equalTo(schema)));
+            return null;
+        });
+
+        pipeline.run();
+    }
+
+    @Test
+    public void testGenerateAttributes() throws IOException {
+
+        String schema =
+                "{"
+                        + "\"id\": \"{{uuid()}}\", "
+                        + "\"eventTime\": \"{{timestamp()}}\", "
+                        + "\"username\": \"{{username()}}\", "
+                        + "\"score\": {{integer(0,100)}}"
+                        + "}";
+        String attributeSchema = "{ \"attribute_test\": \"{{integer(5,100)}}\"}";
+
+
+        File file = tempFolder.newFile();
+        writeToFile(file.getAbsolutePath(), schema);
+
+        File attributeFile = tempFolder.newFile();
+        writeToFile(attributeFile.getAbsolutePath(), attributeSchema);
+
+        // Act
+        //
+        PCollection<PubsubMessage> results =
+                pipeline
+                        .apply("CreateInput", Create.of(0L))
+                        .apply("GenerateMessage", ParDo.of(new MessageGeneratorFn(file.getAbsolutePath(), attributeFile.getAbsolutePath())));
+
+        // Assert
+        //
+        PAssert.that(results)
+                .satisfies(
+                        input -> {
+                            PubsubMessage message = input.iterator().next();
+                            assertThat(message, is(notNullValue()));
+                            assertThat(message.getPayload(), is(notNullValue()));
+                            assertThat(message.getAttributeMap(), is(notNullValue()));
+                            assertThat(message.getAttribute("attribute_test"), is(notNullValue()));
+                            assertThat(Integer.parseInt(message.getAttribute("attribute_test")), is(greaterThan(5)));
```

Comment:
Datagenerator uses random.nextInt(upperbound) + lowerbound to generate random integer values between min and max
https://docs.oracle.com/javase/8/docs/api/java/util/Random.html
https://github.com/vincentrussell/json-data-generator/blob/master/src/main/java/com/github/vincentrussell/json/datagenerator/functions/impl/FunctionUtils.java.
Should compare with greaterThanOrEqualTo

Reply:
Fixed and changed implementation.